# CrunchieMunchies

You work in marketing for a food company <b>myCorps</b>, which is developing a new kind of tasty, wholesome cereal called <b>CrunchieMunchies</b>.

You want to demonstrate to consumers how healthy your cereal is in comparison to other leading brands, so you’ve dug up nutritional data on several different competitors.

Your task is to use NumPy statistical calculations to analyze this data and prove that your <b>CrunchieMunchies</b> is the healthiest choice for consumers.

# Task STEPS

#### 1. First, import numpy.

In [1]:
import numpy as np

#### 2. Look over the cereal.csv file. This file contains the reported calorie amounts for different cereal brands. Load the data from the file and save it as calorie_stats.

In [2]:
calorie_stats = np.genfromtxt('cereal.csv', delimiter=",")
calorie_stats

array([ 70., 120.,  70.,  50., 110., 110., 110., 130.,  90.,  90., 120.,
       110., 120., 110., 110., 110., 100., 110., 110., 110., 100., 110.,
       100., 100., 110., 110., 100., 120., 120., 110., 100., 110., 100.,
       110., 120., 120., 110., 110., 110., 140., 110., 100., 110., 100.,
       150., 150., 160., 100., 120., 140.,  90., 130., 120., 100.,  50.,
        50., 100., 100., 120., 100.,  90., 110., 110.,  80.,  90.,  90.,
       110., 110.,  90., 110., 140., 100., 110., 110., 100., 100., 110.])

#### 3. There are 60 calories per serving of CrunchieMunchies. How much higher is the average calorie count of your competition?

#### Save the answer to the variable average_calories and print the variable to the terminal to see the answer.

In [5]:
average_calories = np.mean(calorie_stats) - 60
print("Average Calories: ", average_calories)

Average Calories:  46.883116883116884


#### 4. Does the average calorie count adequately reflect the distribution of the dataset? Let’s sort the data and see. 

#### Sort the data and save the result to the variable calorie_stats_sorted. Print the sorted data to the terminal.

In [7]:
calorie_stats_sorted = np.sort(calorie_stats)

print("Calorie Stats Sorted", calorie_stats_sorted)

Calorie Stats Sorted [ 50.  50.  50.  70.  70.  80.  90.  90.  90.  90.  90.  90.  90. 100.
 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100.
 100. 100. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 120. 120. 120. 120. 120. 120. 120. 120. 120. 120. 130.
 130. 140. 140. 140. 150. 150. 160.]


#### 5. Do you see what I’m seeing? Looks like the majority of the cereals are higher than the mean. Let’s see if the median is a better representative of the dataset.

#### Calculate the median of the dataset and save your answer to median_calories. Print the median so you can see how it compares to the mean.

In [10]:
median_calories = np.median(calorie_stats_sorted)
print("Median: ", median_calories,",  Mean:",np.mean(calorie_stats))

Median:  110.0 ,  Mean: 106.88311688311688


#### 6. While the median demonstrates that at least half of our values are over 100 calories, it would be more impressive to show that a significant portion of the competition has a higher calorie count that CrunchieMunchies.

#### Calculate different percentiles and print them to the terminal until you find the lowest percentile that is greater than 60 calories. Save this value to the variable nth_percentile.

In [19]:
a = np.arange(1,77)
c = np.percentile(calorie_stats_sorted,a) 
print(c)
#calorie_stats_sorted [calorie_stats_sorted]
nth_percentile = c[c>60].min()

print("\n Lowest percentile that is greater than 60:", nth_percentile,  ", at number:", a[c>60].min())

[ 50.   50.   55.6  70.   70.   75.6  83.2  90.   90.   90.   90.   90.
  90.   90.   90.   91.6  99.2 100.  100.  100.  100.  100.  100.  100.
 100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
 100.  100.  106.4 110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110. ]

 Lowest percentile that is greater than 60: 70.0 , at number: 4


#### 7. While the percentile shows us that the majority of the competition has a much higher calorie count, it’s an awkward concept to use in marketing materials.

#### Instead, let’s calculate the percentage of cereals that have more than 60 calories per serving. Save your answer to the variable more_calories and print it to the terminal

In [20]:
more_calories = np.mean(calorie_stats > 60) * 100
more_calories

96.1038961038961

#### 8. Wow! That’s a really high percentage. That’s going to be very useful when we promote CrunchieMunchies. But one question is, how much variation exists in the dataset? </em></q>Can we make the generalization that most cereals have around 100 calories or is the spread even greater?

#### Calculate the amount of variation by finding the standard deviation Save your answer to calorie_std and print to the terminal. How can we incorporate this value into our analysis?

In [21]:
calorie_std = np.std(calorie_stats_sorted)
calorie_std

19.35718533390827

#### 9. Write a short paragraph that sums up your findings and how you think this data could be used to myCorp’s advantage when marketing CrunchieMunchies.

The findings shows that calories comparison of <b>"CrunchieMunchies"</b> and its competitors cereals. CrunchieMunchies carries less calories than "96%" of other cereals. For other cereals, the "median" calorie count shows "110" and the "mean" at "106.88" having a "standard deviation" of "19.35". In contrast with its competitors' cereals, CrunchieMunchies "calorie count" places it at "4th percentile" leaving only 3 competiotrs having lesser  calories.CrunchieMunchies cereal has "60 calories" which is way less then most of the cereals. It seems CrunchieMunchies will attract huge attention of health conscious consumers, Hence consumers will be interested to buy CrunchieMunchies cereal.